pkg/snapstore/abs_snapstore_test.go

```diff
@@ -7,286 +7,163 @@ package snapstore_test
-	if err != nil {
-		return nil, err
+	// Check if a client was made before for a blob that does not exist yet (contents are staged)
+	if _, ok := c.blobClients[blobName]; ok {
+		return c.blobClients[blobName]
 	}
-	httpReq.ContentLength = request.ContentLength
 
-	httpResp := &http.Response{
-		Request: httpReq,
-	}
-	switch request.Method {
-	case "GET":
-		object := parseObjectNamefromURL(request.URL)
-		if len(object) == 0 {
-			if err := p.handleContainerGetOperation(httpResp); err != nil {
-				return nil, err
-			}
-		} else {
-			p.handleBlobGetOperation(httpResp)
-		}
-	case "PUT":
-		p.handleBlobPutOperation(httpResp)
-	case "DELETE":
-		p.handleDeleteObject(httpResp)
-	default:
-		return nil, err
+	// New client if a client was not made before, or if the snapshot does not exist
+	c.blobClients[blobName] = &fakeBlockBlobClient{name: blobName,
+		deleteFn: func() {
+			delete(c.objects, blobName)
+		},
+		checkExistenceFn: func() bool {
+			_, ok := c.objects[blobName]
+			return ok
+		},
+		commitFn: func(b *[]byte) {
+			c.objects[blobName] = b
+		},
+		getContentFn: func() *[]byte {
+			return c.objects[blobName]
+		},
+		staging: make(map[string][]byte),
 	}
-
-	return pipeline.NewHTTPResponse(httpResp), nil
+	return c.blobClients[blobName]
 }
 
-// handleContainerGetOperation prepares response for Get operation on container.
-func (p *fakePolicy) handleContainerGetOperation(w *http.Response) error {
-	query := w.Request.URL.Query()
-	comp := query.Get("comp")
-	switch comp {
-	case "":
-		w.StatusCode = http.StatusOK
-		w.Body = http.NoBody
-		return nil
-	case "list":
-		return p.handleListObjects(w)
-	default:
-		return nil
-	}
+type fakeBlockBlobClient struct {
+	name             string
+	staging          map[string][]byte
+	mutex            sync.Mutex
+	deleteFn         func()
+	checkExistenceFn func() bool
+	commitFn         func(*[]byte)
+	getContentFn     func() *[]byte
 }
 
-// handleListObjectNames responds with a blob `List` response.
-func (p *fakePolicy) handleListObjects(w *http.Response) error {
-	var (
-		keys          []string
-		blobs         []blobItem
-		query         = w.Request.URL.Query()
-		prefix        = query.Get("prefix")
-		marker        = query.Get("marker")
-		maxResultsStr = query.Get("maxresults")
-		limit         = 1 //Actually for azure its 5000
-		nextMaker     string
-	)
-	if len(maxResultsStr) != 0 {
-		maxResult, err := strconv.Atoi(maxResultsStr)
-		if err != nil {
-			return err
-		}
-		limit = maxResult
-	}
-
-	for k := range p.objectMap {
-		keys = append(keys, k)
-	}
-	sort.Strings(keys)
-	for _, key := range keys {
-		if strings.Compare(key, marker) > 0 {
-			blob := blobItem{
-				Name: key,
-			}
-			blobs = append(blobs, blob)
-			if len(blobs) == limit {
-				nextMaker = key
-				break
-			}
-		}
+// DownloadStream returns the only field that is accessed from the response, which is the io.ReaderCloser to the data
```

Comment:
```suggestion
// DownloadStream returns the only field that is accessed from the response, which is the io.ReadCloser to the data
```
I'm guessing you meant this right?